frontend/src/pages/org/archived-item-qa/ui/replay.ts

```diff
@@ -1,29 +1,73 @@
+import { msg } from "@lit/localize";
+import type { SlRequestCloseEvent } from "@shoelace-style/shoelace";
 import { html } from "lit";
 import { guard } from "lit/directives/guard.js";
 import { when } from "lit/directives/when.js";
 
-import type { ReplayData } from "../types";
+import type { QATab, ReplayData } from "../types";
 
 import { renderSpinner } from "./spinner";
 
+import type { Dialog } from "@/components/ui/dialog";
 import { tw } from "@/utils/tailwind";
 
-export function renderReplay(crawlData: ReplayData) {
+export function renderReplay(crawlData: ReplayData, tab: QATab) {
   return html`
-    <div
-      class=${tw`relative h-full overflow-hidden rounded-b-lg border-x border-b bg-slate-100 p-4 shadow-inner`}
-    >
-      ${guard([crawlData], () =>
-        when(
-          crawlData?.replayUrl,
-          (replayUrl) =>
-            html`<iframe
-              src=${replayUrl}
-              class=${tw`h-full w-full overflow-hidden rounded border bg-neutral-0 shadow-lg`}
-            ></iframe>`,
-          renderSpinner,
-        ),
-      )}
+    <div class="replayContainer ${tw`h-full [contain:paint]`}">
+      <div
+        class=${tw`relative h-full overflow-hidden rounded-b-lg border-x border-b bg-slate-100 p-4 shadow-inner`}
+      >
+        ${guard([crawlData], () =>
+          when(
+            crawlData?.replayUrl,
+            (replayUrl) =>
+              html`<iframe
+                id="interactiveReplayFrame"
+                src=${replayUrl}
+                class=${tw`h-full w-full overflow-hidden rounded border bg-neutral-0 shadow-lg`}
+                @load=${(e: Event) => {
+                  const iframe = e.currentTarget as HTMLIFrameElement;
+
+                  void iframe
+                    .closest(".replayContainer")
+                    ?.querySelector<Dialog>("btrix-dialog.loadingPageDialog")
+                    ?.hide();
+                  /// Prevent anchor tag navigation
+                  iframe.contentDocument?.querySelectorAll("a").forEach((a) => {
+                    a.addEventListener("click", (e: MouseEvent) => {
```

Comment:
@ikreymer seems like wombat.js also attaches an event handler to the anchor tags--this seemed like the simplest solution, but maybe there's a better way to handle this with wombat?

Reply:
wombat shouldn't do that specifically, i think this may be the best option for now, as discussed.